pkg/gateway/model_build_targetgroup.go

```diff
@@ -72,47 +73,38 @@ func (b *targetGroupBuilder) Build(ctx context.Context, srvExport *mcs_api.Servi
 		Client:    b.Client,
 	}
 
-	if err := task.run(ctx); err != nil {
+	if err := task.buildModel(ctx); err != nil {
 		return task.stack, task.targetGroup, corev1.ErrIntOverflowGenerated
 	}
 
 	return task.stack, task.targetGroup, nil
 }
 
-// for serviceexport
-func (t *targetGroupModelBuildTask) run(ctx context.Context) error {
-	/*
-		if !t.serviceExport.DeletionTimestamp.IsZero() {
-			// TODO handle delete
-			return nil
-		}
-	*/
-
-	err := t.buildModel(ctx)
-
-	return err
-}
-
 // for serviceexport
 func (t *targetGroupModelBuildTask) buildModel(ctx context.Context) error {
-	err := t.BuildTargetGroup(ctx)
+	err := t.buildTargetGroupForServiceExport(ctx)
 
 	if err != nil {
 		glog.V(6).Infof("Failed to build TargetGroup when serviceExport buildModel for name %v namespace %v \n", t.serviceExport.Name, t.serviceExport.Namespace)
 		return err
 	}
+	if !t.serviceExport.DeletionTimestamp.IsZero() {
+		//for serviceExport deletion request, we don't need to build targets model
+		return nil
+	}
```

Comment:
for serviceExport deletion request, we don't need to build targets model because the target_group_manager could handle the deregistering targets 
https://github.com/aws/aws-application-networking-k8s/blob/a8eefa1d4c99c6de030f0a6264b5e1a6b10d1b2e/pkg/deploy/lattice/target_group_manager.go#L201

Reply:
I think for create/ update serviceExport, the t.serviceExport.DeletionTimestamp.IsZero() == true? So update serviceExport can update the target?

For delete serviceExport t.serviceExport.DeletionTimestamp.IsZero() == false and we be sure need to de-register all targets for that tg and delelete that target group?